src/main/php/PHPMD/Rule/Naming/ShortVariable.php

```diff
@@ -171,12 +171,59 @@ private function getExceptionsList()
      */
     private function isNameAllowedInContext(AbstractNode $node)
     {
+        if ($this->isChildOf($node, 'ForeachStatement')) {
+            return $this->isInitializedInLoop($node);
+        }
+
         return $this->isChildOf($node, 'CatchStatement')
             || $this->isChildOf($node, 'ForInit')
-            || $this->isChildOf($node, 'ForeachStatement')
             || $this->isChildOf($node, 'MemberPrimaryPrefix');
     }
 
+    /**
+     * Checks if a short name is initialized within a foreach loop statement
+     *
+     * @param \PHPMD\AbstractNode $node
+     * @return boolean
+     */
+    protected function isInitializedInLoop(AbstractNode $node)
+    {
+        $exceptionVariables = [];
```

Comment:
```suggestion
        $exceptionVariables = array();
```
For backward-compatibility (PHPMD is compatible PHP 5.3 => 8)